spec/syntax.md

```diff
@@ -275,6 +275,20 @@ This local variable can then be used in other expressions within the same messag
 declaration = let s variable [s] "=" [s] expression
 ```
 
+Declaration examples:
+
+```
+let $foo = {|A literal value|}
+```
+
+```
+let $foo = {:func opt=$bar}
+```
+
+```
+let $foo = {$foo :number minimumFractionDigits=2}
+```
```

Comment:
Even though it's explained in the formatting doc, is it worth explaining here as well that in the third example, either there must be a previous in-scope definition of `$foo`, or this declaration is syntactically correct but semantically invalid?